### PR TITLE
Do not hide all the elements of the units with problems to anonymous …

### PR DESCRIPTION
**Pull request by morenol**
With this change, the units with problem components are not hidden

image

### CODE REVIEW

File: common/lib/xmodule/xmodule/seq_module.py [hunk 1]
```diff
@@ -465,13 +465,6 @@ def _render_student_view_for_items(self, context, display_items, fragment, view=
             item_type = item.get_icon_class()
             usage_id = item.scope_ids.usage_id
 
-            if item_type == 'problem' and not is_user_authenticated:
```
Comment (cocococosti) [inline common/lib/xmodule/xmodule/seq_module.py]: Just a question: the whole block is classified as "problem" if there is a graded element in it?

Comment (morenol) [reply common/lib/xmodule/xmodule/seq_module.py]: It is being used the icon class to define the type of unit. If there is a component in the unit with 'problem' as icon class then the icon class of the unit is "problem". Those cases are the ones that are failing.

Comment (cocococosti) [reply common/lib/xmodule/xmodule/seq_module.py]: ah yes, they said it in the comments, getting the type from the icon class is kinda weird

**Comment by felipemontoya** [comment]
I suggest that we wait for the upstream PR to be merged and then do a backport before we continue with this. The idea is that we don't deviate too much from upstream here so that we can support this better in the future.

**Comment by morenol** [comment]
**Update**: this is the PR against upstream: https://github.com/edx/edx-platform/pull/22385

**Pull request merged**